suse_caasp

```diff
@@ -332,23 +332,36 @@ find_and_log_files () {
 }
 
 #############################################################
-section_header "Checking etcd service ..."
+section_header "Checking etcd cluster ..."
 OF=etcd.txt
 plugin_message "All data stored in $OF"
 
-set -a
-[ -f /etc/sysconfig/etcdctl ] && source /etc/sysconfig/etcdctl
-set +a
-if check_rpm etcd && [ -e /usr/bin/etcdctl ]; then
-    rpm_verify $OF etcd
-    log_cmd $OF 'etcdctl --version'
-    log_cmd $OF 'etcdctl member list'
-    log_cmd $OF 'etcdctl cluster-health'
-    log_cmd $OF 'etcdctl get /flannel/network/config'
-    log_cmd $OF 'etcdctl ls /flannel/network/subnets'
-    log_cmd $OF 'systemctl status --full etcd.service'
-    log_cmd $OF 'journalctl -u etcd'
-    conf_files $OF /etc/sysconfig/etcd
+if check_rpm curl && [ -e /usr/bin/curl ]; then
+    rpm_verify $OF curl
+    # Run this only on the CaaSP Master nodes
+    if sudo crictl pods | grep etcd &> /dev/null; then
+        # Health check:
+        log_cmd $OF 'curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/server.key --cert /etc/kubernetes/pki/etcd/server.crt https://localhost:2379/health'
+        # ETCD member list
+        log_cmd $OF 'curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/server.key --cert /etc/kubernetes/pki/etcd/server.crt https://localhost:2379/v2/members'
+        # Leader information (avail. only from the leader node)
+        log_cmd $OF 'curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/server.key --cert /etc/kubernetes/pki/etcd/server.crt https://localhost:2379/v2/stats/leader'
+        # Current member information
+        log_cmd $OF 'curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/server.key --cert /etc/kubernetes/pki/etcd/server.crt https://localhost:2379/v2/stats/self'
+        # Statistics
+        log_cmd $OF 'curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/server.key --cert /etc/kubernetes/pki/etcd/server.crt https://localhost:2379/v2/stats/store'
+         # Metrics
+        log_cmd $OF 'curl -Ls --cacert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/server.key --cert /etc/kubernetes/pki/etcd/server.crt https://localhost:2379/metrics'
+    fi
+fi
+
+if check_rpm cri-o && [ -e /usr/bin/crictl ] && [ -S /var/run/crio/crio.sock ]; then
```

Comment:
Can you please move this check and cover `if sudo crictl pods | grep etcd &> /dev/null; the` from line https://github.com/SUSE/supportutils-plugin-suse-caasp/pull/40/files#diff-b0e2fb189cb9a1faffdc4391f6a7643aR342

Reply:
JFYI there might be more than one instances. e.g. after the running the update of the cluster sometimes you end up with:

```bash
crictl pods | grep etcd
f55af15304f00       3 days ago          Ready               etcd-vm072050.qa.prv.suse.net                      kube-system         1
e33fb1d867ed0       4 days ago          NotReady            etcd-vm072050.qa.prv.suse.net                      kube-system         0
```